File: call/ZegoCallDemoOC/ZegoCallDemoOC/ViewController.m
//
//  ViewController.m
//  ZegoCallDemoOC
//
//  Created by zego on 2024/5/27.
//

#import "ViewController.h"
@import ZegoUIKit;
@import ZegoUIKitPrebuiltCall;
@import ZegoPluginAdapter;
@interface ViewController ()
@property (nonatomic, copy) NSString *userName;
@property (nonatomic, copy) NSString *userID;
@property (nonatomic, copy) NSString *appSign;
@property (nonatomic, assign) unsigned int appID;
@property (weak, nonatomic) IBOutlet UILabel *userNameLabel;
@property (weak, nonatomic) IBOutlet UILabel *userIDLabel;
@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view.
  self.userID = [NSString stringWithFormat:@"%u",arc4random() % 99999];
  self.userName = [NSString stringWithFormat:@"zego_%@",self.userID];
  self.appID = <#appID#>;
  self.appSign = <#appSign#>;
}

- (IBAction)makeNewCall:(id)sender {
  self.userID = self.userIDLabel.text.length > 0 ? self.userIDLabel.text : [NSString stringWithFormat:@"%u",arc4random() % 99999];
  self.userName = [NSString stringWithFormat:@"zego_%@",self.userID];
  
  ZegoUIKitPrebuiltCallConfig *config = [ZegoUIKitPrebuiltCallConfig oneOnOneVideoCall];
  ZegoUIKitPrebuiltCallVC *vc = [[ZegoUIKitPrebuiltCallVC alloc] init:self.appID appSign:self.appSign userID:self.userID userName:self.userName callID:@"100" config:config];
  vc.modalPresentationStyle = UIModalPresentationFullScreen;
  [self presentViewController:vc animated:YES completion:nil];
}


@end

File: call/ZegoCallDemo/ZegoCallDemo/ViewController.swift
//
//  ViewController.swift
//  ZegoCallDemo
//
//  Created by zego on 2022/8/19.
//

import UIKit
import ZegoUIKitPrebuiltCall
import ZegoUIKit

class ViewController: UIViewController {
    
    let selfUserID: String = String(format: "%d", Int.random(in: 0...99999))
    var selfUserName: String?
    let yourAppID: UInt32 = <#YourAppID#>
    let yourAppSign: String = <#YourAppSign#>
    
    @IBOutlet weak var userIDLabel: UILabel! {
        didSet {
            userIDLabel.text = selfUserID
        }
    }
    @IBOutlet weak var userNameLabel: UILabel! {
        didSet {
            selfUserName = String(format: "zego_%@", selfUserID)
            userNameLabel.text = selfUserName
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }
    
    @IBAction func makeNewCall(_ sender: Any) {
        
        let config: ZegoUIKitPrebuiltCallConfig = ZegoUIKitPrebuiltCallConfig.oneOnOneVideoCall()
        let callVC = ZegoUIKitPrebuiltCallVC.init(yourAppID, appSign: yourAppSign, userID: selfUserID, userName: self.selfUserName ?? "", callID: "100", config: config)
        callVC.modalPresentationStyle = .fullScreen
        self.present(callVC, animated: true, completion: nil)
    }


}

